packages/contract/README.md

```diff
@@ -310,6 +310,52 @@ const val = instance.getValue();
 // since the contract returns that value.
 ```
 
+#### Overloaded functions
+
+You may find yourself in the situation of having a Truffle contract object that has multiple functions with the same name.
+You can call these "overloaded functions" just like you would a normal contract function. Truffle contract instances
+actually wrap web3's contract abstraction (`web3.eth.Contract`) and so when you call an overloaded function, it
+uses the same function resolution that web3 uses. However, we must give a warning that this overloaded function
+resolution is a bit dodgy and can resolve to the wrong function when you call it. So be careful!
+
+The more unambiguous way of calling these types of functions is to use the `methods` namespace on Truffle contract
+objects. Consider the following contrived code sample:
+
+```solidity
+contract MyContract {
+  uint256 public myUint;
+  string public myString;
+
+  function setValue(uint val) {
+    myUint = val;
+  }
+
+  function setValue(string str) {
+    myString = str;
+  }
+}
+```
+
+This Solidity contract contains two functions named `setValue`, each taking a different type of input. In your JavaScript you
+might do something like the following:
+
+```javascript
+await instance.setValue("this is not string cheese");
+```
+
+Truffle contract (internally using web3's overloaded function resolution) will see that you input a string and
+call the second method in the contract to set `myString`. Again however, sometimes when your contract is more
```

Comment:
Hahaha you wish!  Web3's overload resolution only works reliably when the functions have different numbers of arguments.  The behavior you describe here is obviously what *ought* to actually happen, but not necessarily what actually will.  In reality there's a decent chance that Web3 tries to go with the integer function *anyway* and then just errors on you because the input wasn't an integer like expected.  This is why I keep saying it's so bad, and the primary reason I created Truffle Encoder with the hope of replacing this!

Reply:
Wow, that is terrible.

Reply:
Oh I didn't realize it was primarily based on number of arguments. Bummer, I'll update this blurb here. I wonder how easy it would be to create our own shim/adapter-thingy for this...